docs/adr/0004-translations.md

```diff
@@ -0,0 +1,53 @@
+# 3. Initial Domain Modelling
+
+Date: 2018-09-11
+
+## Context
+
+Every feature we add to the app comes with its own static text, which is either embedded in the code (Ruby or JavaScript) or in the HTML. Static text can be anything from the page title, to the text of a button, to an entire page of guidance.
+
+Writing text 'inline' makes it hard for us to audit all of strings in our application, some of which can only be seen under special conditions e.g. error messages. It also makes it hard to change strings consistently across the application - a task which has to be done by a developer. Finally, using inline strings in code distracts from the logical flow of that code.
+
+[Rails Internationalization](https://guides.rubyonrails.org/i18n.html) (also referred to as 'translations') are a way to extract all of the strings in the application to a central location in `config/locales/en`. The strings can be organized in a hierarchy over one or more files, as below, where we can refer to the reviewed title by writing `I18n.t("publish_document.published.reviewed.title")`.
+
+```
+# publish_document/published.yml
+en:
+  publish_document:
+    published:
+      reviewed:
+        title: Content has been published
+        body: |
+          ‘%{title}’ has been published on GOV.UK.
+
+          It may take 5 minutes to appear live.
+
+```
+
+Rails translations have a few special behaviours, such as pluralization, raw HTML, and variables. The `%{title}` string in the above is an example of a variable, which a developer will set to the title of the document being published.
+
+## Decision
+
+Although we could use translations to extract all of the strings in the application, in some cases we felt this wasn't necessary, or that a different method should be used. The following is a summary of the rules we currently use.
+
+   * **Link and button labels** are not extracted. We think link and button labels are unlikely to change, and extracting them made the application tests harder to read by obfuscating some of the crucial steps in the test with translation keys.
```

Comment:
I think this implies we'd have to test against a translation which may not be unanimous. I'm making a related comment about a testing section below though where I'll expand on this.

Reply:
I've added a testing section now, so this shouldn't be so surprising.